flink-core/src/main/java/org/apache/flink/api/connector/source/lib/util/SplitRequestEvent.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.connector.source.lib.util;
+
+import org.apache.flink.api.connector.source.SourceEvent;
+import org.apache.flink.api.connector.source.SourceReader;
+import org.apache.flink.api.connector.source.SplitEnumerator;
+
+/**
+ * A {@code SourceEvent} representing the request for a split, typically sent from the
+ * {@link SourceReader} to the {@link SplitEnumerator}.
+ */
+public final class SplitRequestEvent implements SourceEvent {
```

Comment:
There is also a `RequestSplitEvent` in `SourceReaderBase`. I am a little unsure if that should be reused. On one hand, it is just an event, the cost of duplicating is little. On the other hand, they are for the exact same purpose, so ideally we want just one class. That means the implementation should depend on `flink-connector-base`, which is doable if we move this source implementation to `flink-connector-testing` as mentioned in the top level comment. 

What do you think?

Reply:
I would not move this to a testing project, because this source is used in non-testing as well.
The predecessor of this is even directly integrated in the DataStream API at the moment: `env.generateSequence()`

Reply:
I will do a follow-up where we consolidate these events.
I think they can be in `flink-core`, because they are just as much needed for sources using the Source API directly, rather than the SplitReader API.